refinery/tool_manager/utils.py

```diff
@@ -300,24 +352,24 @@ def validate_tool_launch_configuration(tool_launch_config):
     with open("tool_manager/schemas/ToolLaunchConfig.json") as f:
         schema = json.loads(f.read())
     try:
-        validate(
-            json.loads(tool_launch_config),
-            schema,
-            resolver=resolver
-        )
-    except ValueError as e:
-        raise RuntimeError(
-            "Tool launch configuration: {} is not valid JSON: {}".format(
-                tool_launch_config,
-                e
-            )
-        )
+        validate(tool_launch_config, schema, resolver=resolver)
     except ValidationError as e:
         raise RuntimeError(
             "Tool launch configuration is not properly configured: {}".format(
                 e
             )
         )
+    # Assert that the data structure being sent over is able to be evaluated
+    #  as a Pythonic Data structure
+    try:
+        nesting = eval(tool_launch_config["file_relationships"])
```

Comment:
We are using `eval()` here because of our current limitations:
- Not being able to store proper JSON in our db due to out of date Django and 3rd party Libraries
- There is no concept of a tuple in the JSON format, so the `()` that represent our `PAIR`s are lost

I have notes to move away from this approach once we have alleviated some technical debt, but for the time being I've implemented error handling as best as I could and have provided full test coverage: 
https://codecov.io/gh/refinery-platform/refinery-platform/pull/1692/src/refinery/tool_manager/utils.py?before=refinery/tool_manager/utils.py#D1-432

https://codecov.io/gh/refinery-platform/refinery-platform/pull/1692/src/refinery/tool_manager/utils.py?before=refinery/tool_manager/utils.py#D1-362